client/src/components/common/JsonSyntaxHighlighter/JsonSyntaxHighlighter.tsx

```diff
@@ -0,0 +1,132 @@
+import { FC, ReactNode, JSX } from 'react';
+
+import styles from './JsonSyntaxHighlighter.module.scss';
+
+type JsonSyntaxHighlighterProps = {
+  json: object | string;
+};
+
+type JsonValue = string | number | boolean | null | JsonObject | JsonArray;
+type JsonObject = { [key: string]: JsonValue };
+type JsonArray = JsonValue[];
+
+export const JsonSyntaxHighlighter: FC<JsonSyntaxHighlighterProps> = ({ json }) => {
+  const jsonObj = typeof json === 'string' ? JSON.parse(json) : json;
+  
+  // Reusable component for comma rendering
+  const Comma = ({ show }: { show: boolean }): JSX.Element | null => 
+    show ? <span className={styles.jsonComma}>,</span> : null;
+  
+  // Reusable wrapper for primitive values
+  const ValueWrapper = ({ 
+    children, 
+    isLastItem 
+  }: { 
+    children: ReactNode; 
+    isLastItem: boolean;
+  }): JSX.Element => (
+    <span>
+      {children}
+      <Comma show={!isLastItem} />
+    </span>
+  );
+  
+  // Component for empty containers ([] or {})
+  const EmptyContainer = ({ 
+    type, 
+    isLastItem 
+  }: { 
+    type: 'array' | 'object';
+    isLastItem: boolean;
+  }): JSX.Element => (
+    <ValueWrapper isLastItem={isLastItem}>
+      <span className={styles.jsonBrace}>
+        {type === 'array' ? '[]' : '{}'}
+      </span>
+    </ValueWrapper>
+  );
```

Comment:
should we put `ValueWrapper` and `EmptyContainer` out of `JsonSyntaxHighlighter`?